lib/ui/src/components/sidebar/SearchResults.tsx

```diff
@@ -127,6 +129,26 @@ const Result: FunctionComponent<
     [onClick]
   );
 
+  const api = useStorybookApi();
+  useEffect(() => {
+    if (props.isHighlighted) {
+      if (item.isComponent) {
+        api.emit(
+          PRELOAD_STORIES,
+          { ids: [item.isLeaf ? item.id : item.children[0]] },
+          {
+            options: {
+              target:
+                item.refId && item.refId !== 'storybook_internal'
+                  ? `storybook-ref-${item.refId}`
+                  : 'storybook-preview-iframe',
+            },
+          }
```

Comment:
We have this same code in 4 places, can we add `api.preloadStories` to `modules/stories.ts` 🙏 ?